### PR TITLE
Make chat prompt input scroll internally

### PR DESCRIPTION
**Pull request by jmagar**
## Summary
- Adds a chat prompt textarea resize helper and keeps the 200px composer cap explicit.
- Switches long prompt overflow to internal textarea scrolling after the cap.
- Resets height, overflow, and scroll position after successful send.

## Verification
- `pnpm --dir apps/gateway-admin exec tsx --test components/chat/chat-input.test.tsx` -> 3/3 pass
- `pnpm --dir apps/gateway-admin run build` -> pass
- `agent-browser` opened `NEXT_PUBLIC_MOCK_DATA=true` dev preview at `http://127.0.0.1:3140/chat/`; preview rendered the chat surface but mock provider disabled the composer, so internal scroll behavior is covered by focused resize tests.

## Notes
- `.env`, `config.toml`, `node_modules`, and `.next` were local ignored worktree support files only and are not committed.

---
## Summary by cubic
Switches long chat prompts to scroll inside the textarea after the 200px cap. Resets size and scroll on send for a smoother compose experience.

- **Bug Fixes**
  - Enable internal textarea scroll after the 200px cap (removes whole-composer overflow).
  - Reset height, overflowY, and scrollTop after a successful send.

- **Refactors**
  - Add `resizeChatPromptTextarea` and constants `CHAT_INPUT_MAX_HEIGHT_PX` (200) and `CHAT_INPUT_MIN_HEIGHT_PX` (44); replace inline resize logic with the helper.
  - Remove textarea `overflow-hidden` class and use constants in styles; add tests for cap behavior and overflow.

<sup>Written for commit 0654d68722da5636c8a8b366aa0818084efc4632. Summary will update on new commits.</sup>

### CODE REVIEW

File: apps/gateway-admin/components/chat/chat-input.test.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    <ChatInput
+      onSend={() => {}}
+      selectedAgent={selectedAgent}
+      agents={[selectedAgent]}
+      onSelectAgent={() => {}}
+    />,
+  )
+
+  assert.match(markup, /aria-label="Message"/)
+  assert.match(markup, /max-height:200px/)
```
[the remark this answers is not in the document]
Comment (jmagar) [reply apps/gateway-admin/components/chat/chat-input.test.tsx]: Fixed in 0654d687

File: apps/gateway-admin/components/chat/chat-input.tsx [hunk 2]
```diff
@@ -217,11 +226,15 @@ export function ChatInput({
...
-            'w-full resize-none overflow-hidden bg-transparent px-3 pt-2.5 pb-1.5 text-[13px] leading-[1.55] sm:px-4 sm:pt-3 sm:pb-2',
+            'w-full resize-none bg-transparent px-3 pt-2.5 pb-1.5 text-[13px] leading-[1.55] sm:px-4 sm:pt-3 sm:pb-2',
             'text-aurora-text-primary placeholder:text-aurora-text-muted/50',
             'outline-none disabled:opacity-50',
           )}
-          style={{ minHeight: '44px', maxHeight: '200px' }}
+          style={{
+            minHeight: '44px',
+            maxHeight: `${CHAT_INPUT_MAX_HEIGHT_PX}px`,
+            overflowY: 'hidden',
```
[the remark this answers is not in the document]
Comment (jmagar) [reply apps/gateway-admin/components/chat/chat-input.tsx]: Fixed in 0654d687
